File: articles/resilience-adaptation-and-long-horizon-decisions/python/run_all_resilience_long_horizon_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for Resilience, Adaptation, and Long-Horizon Decisions."""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "resilience_stock_model.py",
    "adaptive_revision_model.py",
    "shock_absorption_model.py",
    "threshold_review_model.py",
    "long_horizon_strategy_comparison.py",
    "decision_record_exporter.py",
    "resilience_adaptation_long_horizon_simulation.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python resilience long-horizon workflows completed.")


if __name__ == "__main__":
    main()

File: articles/resilience-adaptation-and-long-horizon-decisions/r/run_all_resilience_long_horizon_workflows.R
# run_all_resilience_long_horizon_workflows.R
# Run all R workflows for Resilience, Adaptation, and Long-Horizon Decisions.

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "resilience_adaptation_long_horizon_workflow.R",
  "strategy_profiles.R",
  "scenario_performance.R",
  "threshold_review_tables.R",
  "resilience_summary.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R resilience long-horizon workflows completed.")
